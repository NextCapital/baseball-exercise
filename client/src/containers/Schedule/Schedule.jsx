import map from 'lodash/map';
import React from 'react';
import PropTypes from 'prop-types';

import Game from '../../elements/Game/Game.jsx';

const Schedule = ({ games }) => {
  const content = games.length === 0 ?
    'No games to display' :
    map(games, (game, i) => <Game { ...game } />);

  return (<div className="schedule">{ content }</div>);
};

Schedule.propTypes = {
  games: PropTypes.arrayOf(PropTypes.object)
};

Schedule.defaultProps = {
  games: []
};

export default Schedule;
