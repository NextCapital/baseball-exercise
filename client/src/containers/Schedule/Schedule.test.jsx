import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Schedule from './Schedule';

const renderer = new ShallowRenderer();

describe('Schedule', () => {
  let properties;

  beforeEach(() => {
    properties = {
      games: []
    };
  });

  afterEach(() => {
    properties = null;
  });

  const render = () => {
    renderer.render(<Schedule {...properties } />);
    return {
      component: renderer.getRenderOutput(),
      instance: renderer.getMountedInstance()
    };
  };

  describe('render', () => {
    test('renders a Schedule with the correct properties', () => {
      const { component } = render();
      expect(component).toMatchSnapshot();
    });
  });
});
