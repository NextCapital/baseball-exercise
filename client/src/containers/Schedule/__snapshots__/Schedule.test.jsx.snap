// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule render renders a Schedule with the correct properties 1`] = `
<div
  className="schedule"
>
  No games to display
</div>
`;
